src/Observers/AddressObserver.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Tychovbh\Mvc\Observers;
+
+use Tychovbh\Mvc\Address;
+use Tychovbh\Mvc\Service\AddressLookup\PdokService;
+
+class AddressObserver
+{
+    public function creating(Address $address)
+    {
+        $address = new Address(PdokService::search($address->zipcode, $address->house_number, $address->country));
```

Comment:
this is incorrect you are creating a new adress, instead you should expand (fill) address given as param to this method